pkg/networkserver/downlink.go

```diff
@@ -600,23 +615,23 @@ func (ns *NetworkServer) processDownlinkTask(ctx context.Context) error {
 					if err != nil {
 						return nil, nil, err
 					}
-					if uint(rx1ChIdx) >= uint(len(dev.MACState.CurrentParameters.Channels)) ||
-						dev.MACState.CurrentParameters.Channels[int(rx1ChIdx)] == nil ||
-						dev.MACState.CurrentParameters.Channels[int(rx1ChIdx)].DownlinkFrequency == 0 {
+					if uint(rx1ChIdx) >= uint(len(macState.CurrentParameters.Channels)) ||
+						macState.CurrentParameters.Channels[int(rx1ChIdx)] == nil ||
+						macState.CurrentParameters.Channels[int(rx1ChIdx)].DownlinkFrequency == 0 {
 						return nil, nil, errCorruptedMACState
 					}
-					rx1DRIdx, err := phy.Rx1DataRate(up.Settings.DataRateIndex, dev.MACState.CurrentParameters.Rx1DataRateOffset, dev.MACState.CurrentParameters.DownlinkDwellTime)
+					rx1DRIdx, err := phy.Rx1DataRate(up.Settings.DataRateIndex, macState.CurrentParameters.Rx1DataRateOffset, macState.CurrentParameters.DownlinkDwellTime)
 					if err != nil {
 						return nil, nil, err
 					}
-					rx1Freq := dev.MACState.CurrentParameters.Channels[int(rx1ChIdx)].DownlinkFrequency
+					rx1Freq := macState.CurrentParameters.Channels[int(rx1ChIdx)].DownlinkFrequency
 
 					req := &ttnpb.TxRequest{
 						Class: ttnpb.CLASS_A,
 					}
 					switch {
-					case dev.MACState.QueuedJoinAccept != nil:
-						// Join-accept downlink for Class A/B/C in Rx1/Rx2
+					case dev.PendingMACState != nil:
```

Comment:
Above you use `macState == dev.PendingMACState`, maybe nice to be consistent

Reply:
not applicable anymore